snippets/binarySearch.md

```diff
@@ -0,0 +1,23 @@
+### binarySearch
+
+Use recursion to perform a logarithmic search similar to `.indexOf()` that finds the index of a value within an array. The two differences being 
+1. Operation only works with sorted arrays
+2. Offers a major performance boost when compared to a linear search or `.indexOf()` 
+
+```js
+const binarySearch = (arr, val, start = 0, end = arr.length - 1) => {
+  if (start > end) return -1;
+  const mid = Math.floor((start + end) / 2);
+  const target = arr[mid];
```

Comment:
You might as well not assign `target` and use it right away in the conditions below, makes no difference performance-wise.

Reply:
Yea, But trying to have readability on par with 30 seconds or less. Feel like when user reads the updated description and see snippet they can easily & quickly figure out which things represent which

Reply:
@atomiks made a nice suggestion about this, which would not harm readability and shorten the code just about enough. I like it, what do you think about it?